### PR TITLE
Updated to use SFTP instead of FTPS

### PR DESCRIPTION
**Pull request by JReissmueller**

### CODE REVIEW

File: cwatch.php [hunk 1]
```diff
@@ -908,49 +908,53 @@ private function getMalwareTab($template, $service, $post)
...
+        Loader::loadComponents($this, ['Security']);
 
         // Get cWatch API
         $api = $this->getApi();
         $service_fields = $this->serviceFieldsToObject($service->fields);
 
         if (isset($post['test_ftp'])) {
-            $use_ftps = isset($post['ftps']);
-            $error_message = Language::_('CWatch.!error.ftp' . ($use_ftps ? 's' : '') . '_test', true);
+            $use_sftp = isset($post['sftp']);
```
Comment (tysonphillips) [inline cwatch.php]: This should probably be Html::ifSet

File: cwatch.php [hunk 2]
```diff
@@ -908,49 +908,53 @@ private function getMalwareTab($template, $service, $post)
...
-                    $ftp_options['curlOptions'][CURLOPT_PROTOCOLS] = CURLPROTO_FTPS;
-                    $ftp_options['curlOptions'][CURLOPT_SSL_VERIFYPEER] = true;
-                    $ftp_options['curlOptions'][CURLOPT_USE_SSL] = CURLUSESSL_ALL;
-                    $ftp_options['curlOptions'][CURLOPT_FTPSSLAUTH] = CURLFTPAUTH_DEFAULT;
-                }
+                if ($use_sftp) {
+                    $ftp = $this->Security->create(
+                        'Net',
+                        'SFTP',
+                        [$post['host'], $post['port']]
```
Comment (tysonphillips) [inline cwatch.php]: These should probably be Html::ifSet

File: cwatch.php [hunk 3]
```diff
@@ -908,49 +908,53 @@ private function getMalwareTab($template, $service, $post)
...
+                    );
 
-                $ftp = new Ftp();
-                $ftp->setServer($protocol . $this->Html->ifSet($post['host']));
-                $ftp->setCredentials(
-                    $this->Html->ifSet($post['username']),
-                    $this->Html->ifSet($post['password'])
-                );
-                $ftp->setOptions($ftp_options);
+                    $success = $ftp->login($post['username'], $post['password']) && $ftp->chdir($post['path']);
```
Comment (tysonphillips) [inline cwatch.php]: These should probably be Html::ifSet

File: cwatch.php [hunk 4]
```diff
@@ -908,49 +908,53 @@ private function getMalwareTab($template, $service, $post)
...
+                        'passive' => true,
+                        'port' => $this->Html->ifSet($post['port']),
+                        'timeout' => 30,
+                        'curlOptions' => array(
+                            CURLOPT_PROTOCOLS => CURLPROTO_FTP,
+                        )
+                    );
+                    $protocol = 'ftp://';
+
+                    $ftp = new Ftp();
```
Comment (tysonphillips) [inline cwatch.php:943]: Where is the Ftp class loaded?

File: cwatch.php [hunk 5]
```diff
@@ -970,7 +974,7 @@ private function getMalwareTab($template, $service, $post)
                     'host' => $this->Html->ifSet($post['host']),
                     'port' => $this->Html->ifSet($post['port']),
                     'path' => $this->Html->ifSet($post['path']),
-                    'protocol' => isset($post['ftps']) ? 'FTPS' : 'FTP'
+                    'protocol' => isset($post['sftp']) ? 'SFTP' : 'FTP'
```
Comment (tysonphillips) [inline cwatch.php]: This should probably be Html::ifSet

**Review by tysonphillips — CHANGES_REQUESTED** [review]

File: cwatch.php (hunk 2 above)
Comment (JReissmueller) [reply cwatch.php]: updated

File: cwatch.php (hunk 3 above)
Comment (JReissmueller) [reply cwatch.php]: updated

File: cwatch.php (hunk 4 above)
Comment (JReissmueller) [reply cwatch.php:943]: In a use statement at the top since it is namespaced

File: cwatch.php (hunk 5 above)
Comment (JReissmueller) [reply cwatch.php]: It is a checkbox, I don't care about the value, only that it is set

File: cwatch.php (hunk 1 above)
Comment (JReissmueller) [reply cwatch.php]: It is a checkbox, I don't care about the value, only that it is set

**Pull request merged**